### PR TITLE
Only log CodyInstalled and CodyUninstalled events on installs/uninsta…

### LINKED ISSUE #43: JetBrains: duplicate `CodyInstalled` events
Context https://sourcegraph.slack.com/archives/C059N5FRYG3/p1697568648010619

Here is where we log the `CodyInstalled` event

https://sourcegraph.com/github.com/sourcegraph/jetbrains/-/blob/src/main/java/com/sourcegraph/telemetry/GraphQlLogger.java?L21

We trigger it on

- installation
- user changes server
- user changes access token

One possible resolution step is to persist which server URL has already received this event so that we never send it again for the same server URL.

### PR DESCRIPTION
**Pull request by dadlerj**
…lls. Log settings change events on server and access token changes

Fixes https://github.com/sourcegraph/jetbrains/issues/43

## Test plan

Unfortunately I don't know how to test this and this is a quick, drive-by edit (and hopefully a low risk change). If you aren't comfortable merging as-is, let me know and I can try to set up a dev environment to test.

### CODE REVIEW

**Comment by olafurpg** [comment]
Look at you @dadlerj writing Kotlin 😁 The diff seems safe to merge wrt. not introducing regressions in the functionality of the plugin itself. I can't speak for whether this resolves the analytics issues, I'll leave it to @sourcegraph/jetbrains-plugin or @Strum355 (who has worked most on the telemetry stuff) to approve the PR.

File: src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt [hunk 1]
```diff
@@ -60,14 +55,9 @@ class AccountSettingChangeListener(project: Project) : ChangeListener(project) {
...
-                codyApplicationSettings.isInstallEventLogged = e
-              }
-            } else if (context.accessTokenChanged &&
-                !codyApplicationSettings.isInstallEventLogged) {
-              GraphQlLogger.logInstallEvent(project).thenAccept { e ->
-                codyApplicationSettings.isInstallEventLogged = e
-              }
+              GraphQlLogger.logCodyEvent(project, "settings.serverURL", "changed")
+            } else if (context.accessTokenChanged) {
+              GraphQlLogger.logCodyEvent(project, "settings.accessToken", "changed")
```
Comment (Strum355) [inline src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt:59]: I couldnt find these on a quick check being emitted for vscode, is this an expected deviation?

File: src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt [hunk 2]
```diff
@@ -60,14 +55,9 @@ class AccountSettingChangeListener(project: Project) : ChangeListener(project) {
 
             // Log install events
```
Comment (Strum355) [inline src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt]: Probably need to remove this line now :smile:

Comment (dadlerj) [inline src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt]:
```suggestion
```

File: src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt (hunk 1 above)
Comment (dadlerj) [reply src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt:59]: See some discussion here. I think this is okay! https://github.com/sourcegraph/jetbrains/issues/43#issuecomment-1887944522

**Review by Strum355 — APPROVED** [review]
LGTM, i havent personally interacted with this area much but these changes seem reasonable

**Pull request merged**